### PR TITLE
neutron: Fix neutron-ha-tool initial start (bsc#967098)

### PR DESCRIPTION
**Pull request by vuntz**
neutron-ha-tool is using a file to find the password to use to connect
to OpenStack. However, that file was not created before the
synchronization on sync marks, resulting in the resource possibly
starting on a node where the file is missing.

https://bugzilla.suse.com/show_bug.cgi?id=967098

### CODE REVIEW

**Comment by aplanas** [comment]
+1

**Comment by jsuchome** [comment]
+1. Is Cloud5 also affected?

**Comment by vuntz** [comment]
> Is Cloud5 also affected?

yes

File: chef/cookbooks/neutron/recipes/network_agents_ha.rb [hunk 1]
```diff
@@ -13,11 +13,29 @@
...
+  # FIXME: While the neutron-ha-tool resource agent allows specifying a CA
+  # Certificate to use for SSL Certificate verification, it's hard to select
+  # right CA file as we allow Keystone's and Neutron's to use different CAs.  So
+  # we just rely on the correct CA files being installed in a system wide default
+  # location.
+  file "/etc/neutron/os_password" do
+    owner "root"
+    group "root"
+    mode "0600"
+    content keystone_settings["admin_password"]
```
Comment (nkrinner) [inline chef/cookbooks/neutron/recipes/network_agents_ha.rb]: keystone_settings is not yet defined here. The call from https://github.com/vuntz/crowbar-openstack/blob/bsc%23967098/chef/cookbooks/neutron/recipes/network_agents_ha.rb#L160 needs to get moved here too.

Currently broken, -1

Comment (vuntz) [inline chef/cookbooks/neutron/recipes/network_agents_ha.rb]: Doh, thanks. Fixed.

**Comment by nkrinner** [comment]
+1

**Comment by rhafer** [comment]
+1

**Pull request merged**